plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunctionProcessor.java

```diff
@@ -0,0 +1,209 @@
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.deltalake.DeltaLakeCdfPageSink.CHANGE_TYPE_COLUMN_NAME;
+import static io.trino.plugin.deltalake.DeltaLakeColumnType.REGULAR;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.getParquetMaxReadBlockRowCount;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.getParquetMaxReadBlockSize;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.isParquetOptimizedNestedReaderEnabled;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.isParquetOptimizedReaderEnabled;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.isParquetUseColumnIndex;
+import static io.trino.plugin.deltalake.functions.tablechanges.TableChangesFileSource.CDF_FILE;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.ptf.TableFunctionProcessorState.Finished.FINISHED;
+import static io.trino.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class TableChangesFunctionProcessor
+        implements TableFunctionSplitProcessor
+{
+    private static final int NUMBER_OF_ADDITIONAL_COLUMNS_FOR_CDF_FILE = 2;
+    private static final int NUMBER_OF_ADDITIONAL_COLUMNS_FOR_DATA_FILE = 3;
+    private ConnectorSession session;
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeConfig deltaLakeConfig;
+    private final ParquetReaderOptions parquetReaderOptions;
+    private final List<DeltaLakeColumnHandle> dataFilesColumns;
+    private final List<DeltaLakeColumnHandle> cdfFilesColumns;
+    private TableChangesFileSource source;
+    private DeltaLakePageSource deltaLakePageSource;
+    private List<DeltaLakeColumnHandle> splitColumns;
+    private long version;
+    private long commitTimestamp;
+
+    public TableChangesFunctionProcessor(
+            ConnectorSession session,
+            TrinoFileSystemFactory fileSystemFactory,
+            DeltaLakeConfig deltaLakeConfig,
+            ParquetReaderConfig parquetReaderConfig,
+            TableChangesFunctionTableHandle handle)
+    {
+        this.session = requireNonNull(session, "session is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.deltaLakeConfig = requireNonNull(deltaLakeConfig, "deltaLakeConfig is null");
+        this.dataFilesColumns = requireNonNull(handle, "handle is null").getColumns();
+        ImmutableList.Builder<DeltaLakeColumnHandle> cdfFilesColumns = ImmutableList.builder();
```

Comment:
The builder declaration is not actually needed. It can be inlined one line below.